### DISCUSSION TITLE
Setting up a peering connection from a third party

### DISCUSSION THREAD

**Discussion by pete0emerson** [gruntwork-io/knowledge-base#107] [Help] [answered]
I am wondering if you could help out how I set up a peering connection to my demo vpc from a third party.

The peering is already set up and I have approved the connection in the console. I need to add the routes to my app -vpc. Do I reference the peering module in my app-vpc file? Do I use the external or just the normal peering module? I am running a bit old versions of the terraform-aws-vpc (v5.6).

I am currently just doing this to test but would like to do it in prod eventually. Can I do it the same way we have our mgmt peering like in this example?

https://gruntwork.io/guides/networking/how-to-deploy-production-grade-vpc-aws/#deploy_application_vpcs

**Comment by gruntwork-discussions** [comment]
You must label this issue with at least one of the following labels before it can be answered:

* [r:terraform-aws-service-catalog](https://github.com/gruntwork-io/terraform-aws-service-catalog)
* [r:terraform-aws-vpc](https://github.com/gruntwork-io/terraform-aws-vpc)
* [r:terraform-aws-architecture-catalog](https://github.com/gruntwork-io/terraform-aws-architecture-catalog)
* [r:terraform-aws-asg](https://github.com/gruntwork-io/terraform-aws-asg)
* [r:terraform-aws-beanstalk](https://github.com/gruntwork-io/terraform-aws-beanstalk)
* [r:terraform-aws-cache](https://github.com/gruntwork-io/terraform-aws-cache)
* [r:terraform-aws-ci](https://github.com/gruntwork-io/terraform-aws-ci)
* [r:terraform-aws-ci-pipeline-example](https://github.com/gruntwork-io/terraform-aws-ci-pipeline-example)
* [r:terraform-aws-cis-service-catalog](https://github.com/gruntwork-io/terraform-aws-cis-service-catalog)
* [r:terraform-aws-data-storage](https://github.com/gruntwork-io/terraform-aws-data-storage)
* [r:terraform-aws-ecs](https://github.com/gruntwork-io/terraform-aws-ecs)
* [r:terraform-aws-eks](https://github.com/gruntwork-io/terraform-aws-eks)
* [r:terraform-aws-elk](https://github.com/gruntwork-io/terraform-aws-elk)
* [r:terraform-aws-kafka](https://github.com/gruntwork-io/terraform-aws-kafka)
* [r:terraform-aws-lambda](https://github.com/gruntwork-io/terraform-aws-lambda)
* [r:terraform-aws-load-balancer](https://github.com/gruntwork-io/terraform-aws-load-balancer)
* [r:terraform-aws-messaging](https://github.com/gruntwork-io/terraform-aws-messaging)
* [r:terraform-aws-mongodb](https://github.com/gruntwork-io/terraform-aws-mongodb)
* [r:terraform-aws-monitoring](https://github.com/gruntwork-io/terraform-aws-monitoring)
* [r:terraform-aws-openvpn](https://github.com/gruntwork-io/terraform-aws-openvpn)
* [r:terraform-aws-sam](https://github.com/gruntwork-io/terraform-aws-sam)
* [r:terraform-aws-security](https://github.com/gruntwork-io/terraform-aws-security)
* [r:terraform-aws-server](https://github.com/gruntwork-io/terraform-aws-server)
* [r:terraform-aws-static-assets](https://github.com/gruntwork-io/terraform-aws-static-assets)
* [r:terraform-aws-zookeeper](https://github.com/gruntwork-io/terraform-aws-zookeeper)
* [r:terragrunt](https://github.com/gruntwork-io/terragrunt)
* [r:terratest](https://github.com/gruntwork-io/terratest)
* [r:repo-copier](https://github.com/gruntwork-io/repo-copier)


[Community Guidelines](https://github.com/gruntwork-io/knowledge-base/discussions/1)
[Getting help](https://gruntwork.io/support/)
[Gruntwork documentation](https://www.gruntwork.io)

**Comment by anouarchattouna** [comment, marked as answer]
You can use the [vpc-peering-external](https://github.com/gruntwork-io/terraform-aws-vpc/tree/master/modules/vpc-peering-external) module and I suggest you to have a look at the [vpc-peering-external example](https://github.com/gruntwork-io/terraform-aws-vpc/tree/master/examples/vpc-peering-external).

**Reply by ina-stoyanova** [reply]
The above answer makes sense! I would also add additional bits of context - why do we recommend the external peering connection over the standard one, and also address the direct asks in the question description:

>I need to add the routes to my app -vpc. Do I reference the peering module in my app-vpc file? 

Yes, you could add something like [this](https://github.com/gruntwork-io/terraform-aws-vpc/blob/48c265d72a30304f1cbd079ee578729b126c3d4c/examples/vpc-peering-external/main.tf#L167-L211) in your `.hcl` or `.tf` file where the internal VPC is defined. 

>Do I use the external or just the normal peering module? I am running a bit old versions of the terraform-aws-vpc (v5.6).

In this case it makes sense to use the external VPC peering module since you're trying to connect a third-party to your internal VPC network. The only things that are under your control in this case are the outgoing from your VPC and allowed incoming connections to your VPC, and which ports they're allowed on. This is where [this example](https://github.com/gruntwork-io/terraform-aws-vpc/tree/master/examples/vpc-peering-external#vpc-peering-external-example) comes in handy.

**Reply by yorinasub17** [reply]
Note for Reference Architecture customers:

For the reference architecture, it may be easiest to add a `terragrunt.hcl` file that calls `vpc-peering-external` module directly, using it like a service module, e.g:

```
terraform {
  source = "git::git@github.com:gruntwork-io/terraform-aws-vpc.git//modules/vpc-peering-external?ref=v0.21.1"
}
```
